File: MVC_DiceGame/MVC_DiceGame_ruby/model/DiceGame.rb
require './model/Dice.rb'

module Model
  class DiceGame

    def initialize
      @d1 = Model::Dice.new
      @d2 = Model::Dice.new
    end

    def dice_one_value
      @d1.face_value
    end

    def dice_two_value
      @d2.face_value
    end

    def play
      @d1.roll
      @d2.roll
      @d1.face_value + @d2.face_value == 7
    end

  end
end

File: Component_DiceGame/DiceGame_java/model/DiceGame.java
package model;

public class DiceGame {
  private Dice m_d1;
  private Dice m_d2;
  
  private java.util.ArrayList<IDiceRolledObserver> m_subscribers;
  
  public DiceGame() {
    m_d1 = new Dice();
    m_d2 = new Dice();
    m_subscribers = new java.util.ArrayList<IDiceRolledObserver>();
  }
  
  public void addSubscriber(IDiceRolledObserver a_sub) {
    m_subscribers.add(a_sub);
  }
  
  public boolean play() {
    
    m_d1.roll();
    for(IDiceRolledObserver obs : m_subscribers) {
      obs.DiceRolled(m_d1.getValue());
    }
    try {
      Thread.sleep(2000);
    } catch (Exception e) {
    }
    
    m_d2.roll();
    for(IDiceRolledObserver obs : m_subscribers) {
      obs.DiceRolled(m_d2.getValue());
    }
    try {
      Thread.sleep(2000);
    } catch (Exception e) {
    }
    
    return m_d1.getValue() + m_d2.getValue() == 7;
  }
}